algosdk/future/transaction.py

```diff
@@ -3276,51 +3282,66 @@ def create_dryrun(
             accts.append(txn.sender)
 
             # Add foreign args if they're set
-            if txn.accounts:
+            if hasattr(txn, "accounts"):
```

Comment:
@jdtzmn For my understanding:  Is there a reason to prefer the `hasattr` checks as shown in favor of casting to the specific type (e.g. `appTxn = cast(ApplicationCallTxn, txn)`) after `issubclass` check?

Reply:
Nope! Good catch. I didn't realize it's guaranteed given the previous logic. I'll switch it to cast instead.